prov/verbs/src/verbs_ep.c

```diff
@@ -114,26 +114,18 @@ ssize_t vrb_post_recv(struct vrb_ep *ep, struct ibv_recv_wr *wr)
 	return ret;
 }
 
-void vrb_shutdown_qp_in_err(struct vrb_ep *ep)
+void vrb_shutdown_ep(struct vrb_ep *ep)
 {
-	struct ibv_qp_attr attr;
-	struct ibv_qp_init_attr init_attr;
 	struct fi_eq_cm_entry entry;
 
-	if (!ep || !ep->ibv_qp || ep->ibv_qp->state == IBV_QPS_UNKNOWN || !ep->eq)
-		return;
-
-	memset(&attr, 0, sizeof(attr));
-	memset(&init_attr, 0, sizeof(init_attr));
-	if (ibv_query_qp(ep->ibv_qp, &attr, IBV_QP_STATE, &init_attr))
-		return;
-	if (attr.cur_qp_state != IBV_QPS_ERR)
+	ofi_genlock_held(vrb_ep2_progress(ep)->active_lock);
+	if (!ep || ep->state != VRB_CONNECTED || !ep->eq)
```

Comment:
ep is already used in the line above, no need to check it here.

Reply:
good catch... coverity would have complained about this.  I'll update

Reply:
when I get a chance I will run io500 with PR

Reply:
No performance regression with IO500/DAOS